### PR TITLE
Fix and improve repeat_on_error

### PR DESCRIPTION
**Pull request by filipgolem**

### CODE REVIEW

File: yajsapi/rest/common.ts [hunk 1]
```diff
@@ -34,7 +36,7 @@ export async function repeat_on_error(
   let start_time = Date.now();
   for (let try_num = 1; try_num <= max_tries; ++try_num) {
     if (try_num > 1) {
-      await sleep(Math.min(interval_ms, start_time + max_duration_ms - Date.now()));
+      await sleep(Math.min(interval_ms, start_time + max_duration_ms - Date.now()) / 1000);
```
Comment (filipgolem) [inline yajsapi/rest/common.ts:39]: `sleep(...)` expected milliseconds.

Comment (zakaprov) [reply yajsapi/rest/common.ts:39]: I think you meant seconds. :shrug:

Comment (filipgolem) [reply yajsapi/rest/common.ts:39]: Yes :)

**Review by azawlocki — APPROVED** [review]
Looks good!

I've updated `goth` version required to run integration tests to `^0.4`, since `0.3.x` does not work correctly with `yagna-0.7.x`.

**Comment by shadeofblue** [comment]
closes https://github.com/golemfactory/yajsapi/issues/213

**Comment by shadeofblue** [comment]
closes https://github.com/golemfactory/yajsapi/issues/222

**Pull request merged**